File: frontend/src/types/history.ts
export type HistoryItemType = 'assistant' | 'document' | 'research'

export type HistoryStatus = 'prototype' | 'draft' | 'completed'

export type HistorySort = 'recent' | 'oldest' | 'title'

export interface AssistantMetadata {
  kind: 'assistant'
  sessionTopic: string
  messageCount: number
}

export interface DocumentMetadata {
  kind: 'document'
  documentType: string
  targetRoute: string
}

export interface ResearchMetadata {
  kind: 'research'
  topics: string[]
  resultCount: number
}

export type HistoryMetadata =
  | AssistantMetadata
  | DocumentMetadata
  | ResearchMetadata

export interface HistoryItem {
  id: string
  type: HistoryItemType
  title: string
  description: string
  createdAt?: string
  updatedAt: string
  status: HistoryStatus
  relatedRoute: string
  category: string
  metadata?: HistoryMetadata
}

export interface HistoryFilter {
  type: HistoryItemType | 'all'
  status: HistoryStatus | 'all'
  sort: HistorySort
}

export interface HistorySummary {
  totalItems: number
  assistantSessions: number
  documents: number
  researchTopics: number
}

/** Backend API DTOs */
export interface HistoryItemResponse {
  item_id: string
  type: HistoryItemType
  title: string
  description: string
  category: string
  status: HistoryStatus
  updated_at: string
  related_route: string
  prototype: boolean
}

export interface HistoryListResponse {
  count: number
  items: HistoryItemResponse[]
  prototype: boolean
}

File: backend/app/schemas/history.py
"""History request/response schemas."""

from enum import Enum
from typing import List

from pydantic import BaseModel, Field


class HistoryItemType(str, Enum):
    """Prototype activity kinds."""

    ASSISTANT = "assistant"
    DOCUMENT = "document"
    RESEARCH = "research"


class HistoryStatus(str, Enum):
    """Prototype activity statuses."""

    PROTOTYPE = "prototype"
    DRAFT = "draft"
    COMPLETED = "completed"


class HistoryItemResponse(BaseModel):
    """Single prototype workspace activity record."""

    item_id: str
    type: HistoryItemType
    title: str
    description: str
    category: str
    status: HistoryStatus
    updated_at: str
    related_route: str
    prototype: bool = True


class HistoryListResponse(BaseModel):
    """Prototype activity list payload."""

    count: int
    items: List[HistoryItemResponse] = Field(default_factory=list)
    prototype: bool = True
